src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1591,7 +1626,14 @@ function fillcombinations(df::AbstractDataFrame, indexcols;
     end
 
     # keep only columns from the source in their original order
-    return select!(out_df, _names(df))
+    select!(out_df, _names(df))
+
+    _copy_metadata!(out_df, df)
+    for i in 1:ncol(out_df)
+        _copy_colmetadata!(out_df, i, df, i)
+    end
```

Comment:
I wanted to avoid defining `_copy_colmetadata!(out_df, df)` as using it could lead to hard to catch bugs if it is used in a place where we are not sure that column names match (directly or after renaming).

Reply:
Would checking that names are equal add a significant overhead? Otherwise the function could check that and that would be safer. But the current loop is OK if the function idea is problematic.

Reply:
OK - I will create a separate function for this and give it a name that makes sure it is clear when it is safe to use it.

Reply:
Done